### ISSUE TITLE
oci_instance_configuration doesn't provide a way to specify boot volume disk size

### ISSUE THREAD

**Issue by skydreamerr** [oracle/oci-ansible-modules#55]
Through oci_instance I can specify "boot_volume_size_in_gbs"
But for instance pool there is no such parameter in instance pool configuration.

**Comment by manojmeda** [comment]
@skydreamerr  The OCI ansible modules depend on OCI SDK/API and the API does not support this option yet. We will add it to the module once the API supports it. Refer https://docs.cloud.oracle.com/iaas/api/#/en/iaas/20160918/datatypes/InstanceConfigurationInstanceSourceViaImageDetails

**Comment by nalsaber** [comment]
Please use the boot volume size "boot_volume_size_in_gbs" in oci_instance module then use that to create an Instance Configuration Source then reference it in the instance pool configuration 'instance_configuration_id'

**Comment by skydreamerr** [comment]
"then use that to create an Instance Configuration Source"
by "that" did you mean boot volume?
I'll try that

**Comment by nalsaber** [comment]
@skydreamerr  Any luck with that?

**Comment by skydreamerr** [comment]
@nalsaber Not yet. I found workaround for my use case by creating image of desired size. But image size is limited.

**Comment by VAdamec** [comment]
```
oci_instance:
...
 source_details:
    source_type: image
    image_id: '{{ oci_image_ocid }}'
    boot_volume_size_in_gbs: '{{ oci_boot_volume|int }}'
...
```

`
--extra-vars 'oci_vmname=demo-test oci_boot_volume=200'
`

and getting this error:

`"Field source_details.boot_volume_size_in_gbs with value 120 was expected to be of type int but was of type str"`


```
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "master:/usr/lib/python2.7/site-packages/ansible/module_utils/oracle/oci_utils.py", line 980, in create_resource
    resource = to_dict(call_with_backoff(create_fn, **kwargs_create).data)
  File "master:/usr/lib/python2.7/site-packages/ansible/module_utils/oracle/oci_utils.py", line 1936, in call_with_backoff
    return fn(**kwargs)
  File "/usr/lib/python2.7/site-packages/oci/core/compute_client.py", line 2257, in launch_instance
    response_type="Instance")
  File "/usr/lib/python2.7/site-packages/oci/retry/retry.py", line 271, in make_retrying_call
    return func_ref(*func_args, **func_kwargs)
  File "/usr/lib/python2.7/site-packages/oci/base_client.py", line 208, in call_api
    body = self.sanitize_for_serialization(body)
  File "/usr/lib/python2.7/site-packages/oci/base_client.py", line 463, in sanitize_for_serialization
    sanitized_dict[key] = self.sanitize_for_serialization(val, value_declared_type, inner_field_name)
  File "/usr/lib/python2.7/site-packages/oci/base_client.py", line 463, in sanitize_for_serialization
    sanitized_dict[key] = self.sanitize_for_serialization(val, value_declared_type, inner_field_name)
  File "/usr/lib/python2.7/site-packages/oci/base_client.py", line 415, in sanitize_for_serialization
    self.raise_type_error_serializing_model(field_name, obj, declared_type)
  File "/usr/lib/python2.7/site-packages/oci/base_client.py", line 471, in raise_type_error_serializing_model
    raise TypeError('Field {} with value {} was expected to be of type {} but was of type {}'.format(field_name, str(obj), declared_type, type(obj).__name__))

fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            .....
            "source_details": {
                "boot_volume_size_in_gbs": "200",
                "image_id": "ocid.xxxxx
                "source_type": "image"
            },
            .....
        }
    },
    "msg": "Field source_details.boot_volume_size_in_gbs with value 200 was expected to be of type int but was of type str"
}

```

**Comment by nalsaber** [comment]
@VAdamec did you try passing an integer for boot_volume_size_in_gbs instead of a string?

**Comment by VAdamec** [comment]
You mean like passing it in JSON format like `--extra-vars '{"oci_vmname":"demo-test","oci_boot_volume":"200"}'` ? Doesn't work also this should works to define type `'{{ oci_boot_volume|int }}'`  right ? seems like some internal stuff in module variable handling.

Btw Ansible 2.7.10-1 and oci ansible modules 1.9.0.

**Comment by nalsaber** [comment]
Did you try passing the size like this without quotes: --extra-vars '{"oci_vmname":"demo-test","oci_boot_volume":200}'

**Comment by skydreamerr** [comment]
@VAdamec @nalsaber All "{{ ... }}" templates produce strings in Ansible. But there is workaround for that put "jinja2_native = True" in Ansible.cfg

**Comment by VAdamec** [comment]
with `ANSIBLE_JINJA2_NATIVE=True` it's ok

**Comment by skydreamerr** [comment]
It seems OCI python sdk now supports boot volume size for instance configuration.
https://oracle-cloud-infrastructure-python-sdk.readthedocs.io/en/latest/api/core/models/oci.core.models.InstanceConfigurationInstanceSourceViaImageDetails.html#oci.core.models.InstanceConfigurationInstanceSourceViaImageDetails

Are there any plan to support it in Ansible?

**Comment by nalsaber** [comment]
Right now the boot_volume_size is missing from our Ansible `instance_configuration` module. 
instance_details -> launch_details -> source_details -> (`boot_volume_size_in_gbs` not supported yet)
It will be added in few months. Can you use any work around for now?

**Comment by manojmeda** [comment]
@skydreamerr Even though not documented, the ansible module should support the parameter `boot_volume_size_in_gbs`. Can you please give it a try.

PS: Upgrade oci to the version which supports this option

**Comment by nalsaber** [comment]
The missing features (`boot_volume_size_in_gbs`) are now available in release [v1.10.0](https://github.com/oracle/oci-ansible-modules/releases/tag/v1.10.0)

**Comment by skydreamerr** [comment]
@manojmeda It works now. Thank you!
